src/rational/mat_q/gso.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright © 2023 Marcel Luca Schmidt
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! Implementation of the Gram-Schmidt Orthogonalization for ['MatQ'] matrices.
+
+use super::MatQ;
+use crate::traits::{GetNumColumns, GetNumRows};
+use flint_sys::fmpq_mat::fmpq_mat_gso;
+
+impl MatQ {
+    /// Computes the Gram-Schmidt Orthogonalization of the matrix and returns a [`MatQ`] with the corresponding matrix.
+    ///
+    /// # Example
+    /// ```
+    /// use qfall_math::rational::MatQ;
+    /// use std::str::FromStr;
+    ///
+    /// let mat = MatQ::from_str("[[1/2, 1],[2, 5]]").unwrap();
+    /// let mat_gso = mat.gso();
+    /// ```
+    pub fn gso(&self) -> Self {
+        let mut out = MatQ::new(self.get_num_rows(), self.get_num_columns()).unwrap();
+        unsafe {
+            fmpq_mat_gso(&mut out.matrix, &self.matrix);
+        }
+        out
+    }
+}
+
+#[cfg(test)]
```

Comment:
I would like your tests to include vec_i.dot_product(&vec_j) != 0 in cases where you would expect it, and one where where you actually check how the orthogonalized matrix is what you would expect. Otherwise the function as of right now could just return a zero matrix in the correct dimension and pretty much pass all tests (except for one with [[1]])